File: process_logo.py
import cv2
import numpy as np

def process_logo():
    # Load image
    img = cv2.imread(r"c:\Users\Spencer\Desktop\darcy\info-agência\Logo Laercom.png", cv2.IMREAD_UNCHANGED)
    
    # Check if it has alpha channel, if not, add it
    if img.shape[2] == 3:
        img = cv2.cvtColor(img, cv2.COLOR_BGR2BGRA)
        
    # Create a mask for the background. The background is mostly white/light gray.
    # Let's use a threshold on the grayscale image to find the background.
    gray = cv2.cvtColor(img, cv2.COLOR_BGRA2GRAY)
    _, mask = cv2.threshold(gray, 230, 255, cv2.THRESH_BINARY)
    
    # We want to keep everything that is NOT background
    # Let's do some morphological operations to clean the mask if needed, but simple threshold might work.
    
    # Create the light mode version (transparent background)
    light_logo = img.copy()
    light_logo[mask == 255, 3] = 0 # Set alpha to 0 for background pixels
    
    cv2.imwrite(r"c:\Users\Spencer\Desktop\darcy\assets\logo-light.png", light_logo)
    
    # Create the dark mode version
    # Dark blue is approx B: 80, G: 30, R: 10 (Wait, it's #081B3C approx, very dark blue)
    # We want to change the dark blue to white.
    # Let's find dark pixels
    dark_logo = light_logo.copy()
    
    # Find pixels where the value is dark (i.e. all B, G, R are less than some threshold)
    # The orange part is bright in Red and Green.
    b, g, r, a = cv2.split(dark_logo)
    
    # A pixel is dark blue if it's generally dark
    # Let's threshold based on brightness
    # Dark blue text has low R, G, and B. Orange has high R, G, low B.
    # So if R < 100, G < 100, B < 150, it's likely the dark blue part.
    dark_mask = (r < 100) & (g < 100) & (b < 150) & (a > 0)
    
    # Set those dark blue pixels to white
    dark_logo[dark_mask, 0] = 255 # B
    dark_logo[dark_mask, 1] = 255 # G
    dark_logo[dark_mask, 2] = 255 # R
    
    cv2.imwrite(r"c:\Users\Spencer\Desktop\darcy\assets\logo-dark.png", dark_logo)
    print("Logos saved.")

if __name__ == '__main__':
    process_logo()

File: process_logo.js
const { Jimp } = require('jimp');

async function processLogo() {
  const image = await Jimp.read('info-agência/Logo Laercom.png');
  const lightLogo = image.clone();
  
  // Make white/light-gray background transparent
  // and REMOVE the "CONSULTORIA EM E-COMMERCE" text
  lightLogo.scan(0, 0, lightLogo.bitmap.width, lightLogo.bitmap.height, function(x, y, idx) {
    const r = this.bitmap.data[idx + 0];
    const g = this.bitmap.data[idx + 1];
    const b = this.bitmap.data[idx + 2];
    
    // Subtitle is at x >= 535 and y >= 435
    if (x >= 535 && y >= 435) {
      this.bitmap.data[idx + 3] = 0; // make transparent
      return;
    }
    
    // If pixel is light gray/white, make transparent
    if (r > 220 && g > 220 && b > 220) {
      this.bitmap.data[idx + 3] = 0; // alpha = 0
    }
  });
  
  // Crop transparent edges
  lightLogo.autocrop();
  await lightLogo.write('assets/logo-light.png');
  
  // Create dark logo
  const darkLogo = lightLogo.clone();
  
  // Convert ONLY the "LAERCOM" text to white
  // Note: Since we already autocropped, the coordinates changed!
  // So we just change any dark pixel to white IF it's on the right side of the image.
  // Let's scan to find where the gap is in the cropped image.
  const w = darkLogo.bitmap.width;
  const h = darkLogo.bitmap.height;
  
  // The icon is roughly the left half or 40% of the image.
  // Let's find the horizontal gap dynamically or just assume x > 300
  let colHasPixel = new Array(w).fill(false);
  darkLogo.scan(0, 0, w, h, function(x, y, idx) {
    if (this.bitmap.data[idx + 3] > 0) colHasPixel[x] = true;
  });
  
  let gapX = 0;
  for(let i = Math.floor(w * 0.2); i < w; i++) {
    if (!colHasPixel[i] && colHasPixel[i-1]) {
      // Found the start of a gap! Let's say the gap is at least 10px
      let isGap = true;
      for(let j = 0; j < 10; j++) {
        if(colHasPixel[i+j]) isGap = false;
      }
      if (isGap) {
        gapX = i + 5;
        break;
      }
    }
  }
  if (gapX === 0) gapX = 260; // fallback
  
  darkLogo.scan(0, 0, w, h, function(x, y, idx) {
    // Only process pixels in the text region
    if (x > gapX) {
      const a = this.bitmap.data[idx + 3];
      // Since it's already cropped, the text is here.
      // Make non-transparent pixels white
      if (a > 0) {
        this.bitmap.data[idx + 0] = 255;
        this.bitmap.data[idx + 1] = 255;
        this.bitmap.data[idx + 2] = 255;
      }
    }
  });
  
  await darkLogo.write('assets/logo-dark.png');
  console.log("Logos processed and cropped properly!");
}

processLogo();
